### ISSUE TITLE
Make the process fail if the changelog file is not found

### ISSUE THREAD

**Issue by MangelMaxime** [ionide/KeepAChangelog#12]
Hello @baronfel,
this is an issue created from our discussion on F# slack.

Currently, if the changelog file is not found there is no indication for the user. The `dotnet pack` command succeed as if nothing was wrong.

I personally think that if the changelog file is not found at the specified path, the process should fail because it is generating something from an invalid context.

For example, when we specify the `PackageLicenseFile` node in an `fsproj` the task is failing if the provided value is invalid or the file is missing/not found.

**Comment by MangelMaxime** [comment]
Thank you for the clear explanations.

Personally, I always preferred explicit behaviour from my code and so would prefer the second option.

**Comment by baronfel** [comment]
I agree. A few other people have spoken up as well. We should remove the implicit mode and provide a helpful error if the user doesn't specify a `ChanglogFile`. That way there's no confusion about if something is happening or not.

**Comment by MangelMaxime** [comment]
Looking at the code it seems like there are already severals checks done for the file existence:

https://github.com/ionide/KeepAChangelog/blob/7ad0461e29bc3f22aafe971cdb11d61451198531/src/Ionide.KeepAChangelog.Tasks/buildMultiTargeting/Ionide.KeepAChangelog.Tasks.targets#L24-L28

https://github.com/ionide/KeepAChangelog/blob/7ad0461e29bc3f22aafe971cdb11d61451198531/src/Ionide.KeepAChangelog.Tasks/Library.fs#L49-L54

I guess they are currently only generating warnings and we should hard fail in these cases. What would be the correct way of failing?

- In F#, we can probably throw an exception
- In MSBuild ?

**Comment by baronfel** [comment]
@tboby just added two msbuild warnings verifying that the changelog file is provided and exists.

**Comment by tboby** [comment]
So, now that it's no longer implicit, what behaviour do we want?

I suspect that the `WarnOnNoChangelogFileExists` (when the given file isn't found) should possibly be an error now.

Do we want the `WarnOnNoChangelogFileSet` to be an error as well, perhaps mentioning "Please set this property ... , or remove the Ionide.KeepAChangelog.Tasks package"?

Alternatively, do we want to add a property to supress the validation? Trying to think if it would be annoying if you Build.props this package and then your test project starts throwing errors.
